modules/userdirectory-brightspace/src/main/java/org/opencastproject/userdirectory/brightspace/BrightspaceUserProviderInstance.java

```diff
@@ -0,0 +1,274 @@
+   * @see org.opencastproject.security.api.UserProvider#getName()
+   */
+  @Override
+  public String getName() {
+    return pid;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getUsers()
+   */
+  @Override
+  public Iterator<User> getUsers() {
+    return Collections.emptyIterator();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#loadUser(java.lang.String)
+   */
+  @Override
+  public User loadUser(String userName) {
+    this.loadUserRequests.incrementAndGet();
+    logger.debug("getting user from cache");
+
+    try {
+      Object user = this.cache.getUnchecked(userName);
+      if (user != this.nullToken) {
+        logger.debug("Returning user {} from cache", userName);
+        return (User) user;
+      } else {
+        logger.debug("Returning null user from cache");
+        return null;
+      }
+    } catch (ExecutionError | UncheckedExecutionException ee) {
+      logger.warn("Exception while loading user {}", userName, ee);
+      return null;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#countUsers()
+   */
+  @Override
+  public long countUsers() {
+    return 0L;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getOrganization()
+   */
+  @Override
+  public String getOrganization() {
+    return this.organization.getId();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#findUsers(java.lang.String, int, int)
+   */
+  @Override
+  public Iterator<User> findUsers(String query, int offset, int limit) {
+    return Collections.emptyIterator();
+  }
+
+  @Override
+  public void invalidate(String userName) {
+    this.cache.invalidate(userName);
+  }
+
+  @Override
+  public Iterator<Role> getRoles() {
+    return Collections.emptyIterator();
+  }
+
+  @Override
+  public List<Role> getRolesForUser(String username) {
+    User user = this.loadUser(username);
+    if (user != null) {
+      logger.debug("Returning cached role set for {}", username);
+      return new ArrayList<>(user.getRoles());
+    }
+    logger.debug("Return empty role set for {} - not found in Brightspace", username);
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Iterator<Role> findRoles(String query, Target target, int offset, int limit) {
+    return Collections.emptyIterator();
+  }
+
+  private User loadUserFromBrightspace(String username) {
+    if (this.cache == null) {
+      throw new IllegalStateException("The Brightspace user detail service has not yet been configured");
+    } else if (!"admin".equals(username) && !"".equals(username) && !"anonymous".equals(username)) {
+
+      JaxbOrganization jaxbOrganization = JaxbOrganization.fromOrganization(organization);
+
+      this.brightspaceWebServiceRequests.incrementAndGet();
+      Thread currentThread = Thread.currentThread();
```

Comment:
Honestly no idea, the code was developed by an external consultant and it seems that this was copy-pasted from the sakai userprovider.

Reply:
I looked into why that's there in the Sakai provider, and can't find an answer - it's been there for quite some time and I can't find the commit or issue that added it.

Reply:
@smarquard I don't see anything obvious regarding why this is needed.  Can you try yanking the loader at your end and see if it breaks anything.  @flyapen, if you have a system to test this against, can you yank it here too and see if it breaks things?

Reply:
I can try on our test system, but not keen to make a possibly-breaking-change in production right now especially as there's no particular problem that we're trying to solve. I think it may be related to caching or something but it would have been done like that in 1.x days.

Reply:
It turns out the Sakai provider was copied from the Ldap provider, and we find this classloader switching there:

https://github.com/opencast/opencast/blob/ceb51e5a42d21997f478b0d84ef674ab4fc310ee/modules/userdirectory-ldap/src/main/java/org/opencastproject/userdirectory/ldap/LdapUserProviderInstance.java#L320

Reply:
Oh man, old code is old.  Now that I look at how this is being used by the LDAP (and Sakai) code, I'm guessing this switching is more or less required.  The factory which creates this provider has the relevant Karaf service bits and hooks into the security system, I think.  It might be worth testing in a test environment to see if this is still required, if you get a chance.  It's always nice to be able to remove black magic from our code :)